### ISSUE TITLE
[Bug] Mouse is jumping

### ISSUE THREAD

**Issue by ghost** [input-leap/input-leap#300]
> This issue has been migrated from old Barrier Github repository https://github.com/debauchee/barrier/issues/300
> 
> Issue created on: 2019-05-07 by @domske
> Issue last updated on: 2021-10-17

#### Bug Report

The mouse is jumping if moved from rest. No problem when I move it continuously.
But if you wait between movements (just a few milliseconds / < second), the mouse jumps.
It's like it sleeps immediately, as soon as she stops moving.

See screen record:
![screenrecord](https://user-images.githubusercontent.com/28209992/57329434-0a716900-7114-11e9-88ce-5b99ec1a35bf.gif)

Reproduce: 
Move the mouse back and forth. While she has come to a complete halt for a moment.

This is very annoying. I expected a smooth movement. This is not possible if the mouse awakes delayed and jumps to position.
Maybe this is a wanted behavior to hibernate the transmission. If so, the timeout time should be increased to a minute or a few seconds.

#### Version
Server: Windows 10 Home (2.2.0)
Client: Mac OS X High Sierra (2.10) (Issue)

Any questions? I am glad to help.

---

> Commented on: 2019-05-08 by @noisyshape

Would I be correct in assuming that you're connecting over wifi? And that one or both systems have a power saving feature enabled? A mitigation for this might be sending dummy packets for a while after mouse movement.

---

> Commented on: 2019-05-08 by @domske

Yes, both devices are connected via WiFi in same network on a router. Do you mean, using LAN will not have this problem?...

---

> Commented on: 2019-05-08 by @AdrianKoshka

> Do you mean, using LAN will not have this problem?...

It's entirely possible, I was tired when I saw this issue, but thanks for @noisyshape for pointing out the possible WiFi problem. In my experience with WiFi and barrier, things do seem latent. I'm not going to say it will fix your issue, but if you could try to reproduce this over ethernet, it'd be appreciated.

---

> Commented on: 2019-05-09 by @mrwensveen

I also have this issue when I'm connected over WiFi. It's not consistent, though. Sometimes everything is silky smooth and sometimes not. I think it might depend on how much the network is utilized (it definitely seems worse when the kids are home and streaming video etc.)

---

> Commented on: 2019-06-28 by @AdrianKoshka

New release, unlikely it fixes this issue, but I thought I'd bring it up: https://github.com/debauchee/barrier/releases/tag/v2.3.0

---

> Commented on: 2019-10-29 by @maliesa96

I had this issue. Decreasing the polling frequency on my mouse fixed it.

---

> Commented on: 2020-03-01 by @tinybot-ca

I recall this being an issue with Synergy way back in the day -- and connecting my client machine (Macbook Pro laptop) via Ethernet got around it. Now, many years later, this still seems to be an issue on Wi-Fi? I suppose I can dig up an ethernet dongle -- but would love for a solution that would work for Wi-Fi.

Update: I connected my MacBook Pro to ethernet via dongle, and sure enough, this lagging/stuttering/jumping went completely away. If you do this, don't forget to stop and reload Barrier to pick-up and use the IP address used by your ethernet connection.

---

> Commented on: 2020-08-07 by @pinei

It seems to perform better over 5 Ghz Wifi than 2 Ghz

---

> Commented on: 2020-10-28 by @mrpew

Bug still relevant, I get frequent mouse stuttering/jumping between two Ubuntu Systems connected via **Ethernet** with a sub 1ms ping. 

Sometimes happens occasionally (e.g. every 10-30 seconds) sometimes constantly (Unusable until barrier process is killed or direct input on the client is given).

While its happening, using the keyboard does not work either.

Happens with both the v2.3.3 release tag as well as the current master branch.

---

> Commented on: 2020-10-30 by @fiftyfathoms

This bug affects me too. Kubuntu host 18.04.5, Windows 10 guest. It's strange that everything used work just fine for months and all of a sudden the mouse pointer started to show extreme stutters today.

---

> Commented on: 2020-11-16 by @Skrimpton

I have the same issue on one manjaro nibia 20.2 client.

Mouse also stops even with continous movement.

It is not an issue (at all) on a clean-installed Manjaro minimal nibia 20.2 laptop, where both share a windows 10 host.

The client with the bug is an upgraded system, that has been in use for about  6/7 months, with several "extra" apps installed.

The bug occurs with both a wireless logitech g602 mouse and a wired hp M-BT96a. Keyboard (logitech k350 wireless) is also affected, but with less noticable lag.

Added: Host pc uses a wired connection, whereas both clients are connected via WI-FI.

Second addition: Seems like rebooting into a "fresh" session fixed whatever was causing the issue for me. Also forgot to add I am running KDE Plasma on both clients.

Sorry if this then is a divergent/different issue, but leaving the comment for others to delete if neccesary 

---

> Commented on: 2020-12-03 by @clankill3r

Same issue with 10.14.6. Ethernet cable made it better, but that's not really an option for me.

---

> Commented on: 2020-12-03 by @fiftyfathoms

An update to my previous comment:
After also checking Synergy and experiencing the same issue, I changed the the wifi band of my router from 2.4 to 5 GHz and the issue seems to be gone. I had switched off the 5 GHz channel long before, mainly due to the rather long distance between my computer and the router. Apparently even slight change in the position of the router helps a great deal with the transmission performance of the signal. 

---

> Commented on: 2020-12-15 by @ishridharhegde

I still have this issue using Windows 10 -> Linux Lite connected over wifi. 
Many of you have suggested to switch to 5GHz band but my router does not have that band!
Is there any other fixes that any of you have found?

When I am using Windows 10 -> Windows 10, Mouse Without Borders works flawlessly !

---

> Commented on: 2021-01-22 by @hungerregnuh

For my setup:
  - Client is a MBP (Catalina and Big Sur)
  - Linux Mint as my server and had no issues
  - I'm now running it as a server on a MBP (Big Sur) and this is when i noticed the mouse jumping issue.

What ultimately fixed it for me was getting my multi-display issue fixed.  Anyone running a MBP with multi-display knows how finicky OSX has been lately with them.  I have a 3 display setup and getting the third display to be stable was key.  Once i got the third display to stop reconnecting the mouse jumping issue went away.  Even when i don't have the third display connected the mouse jumps.  I suspect barrier, OSX or both are "remembering" the third display even though it's not connected and is causing the mouse to jump around as something is constantly changing the actual width of my desktops.

not sure if this helps, hope it does.

---

> Commented on: 2021-02-02 by @hungerregnuh

Not a fix, but it works - i also found out that if you enable the double tap edge to switch, the mouse doesn't jump around anymore, not as convenient sure - but works.

---

> Commented on: 2021-02-26 by @LeonardSSH

Has anyone solved this problem? (🟦Windows > 🐧Ubuntu) 😊

---

> Commented on: 2021-03-05 by @bbook1997

Running Windows 10 Pro 10.0.19041 Build on two PC's.   Connected via wireless for months now.
Running Barrier 2.3.3-release-f2bf7cb7 without any issues during all this time except for one time with mouse became sluggish on client.   Restarting the server fixed it.   Been running fine for weeks since then.
Suddenly, yesterday, the mouse has become very choppy on the client.   I've rebooted both server and client twice since then, and it is still the same.  I tried @hungerregnuh 's workaround, but that didn't help.  I've tried changing the port used, but that hasn't helped.   Logging is not turned on.   Over 100GB available on all disk drives.  
Would love to get some ideas on how to fix it.

I believe it is related to the fact that I have been doing a lot of testing using applications that talk to each other over the same network, and yesterday I had them sending tons of messages back and forth.   But that test is not running now, and after rebooting, I had hoped it would clear out whatever is causing the delays, but nothing has helped so far.

Edit:
After uninstalling on both client and server and reinstalling, it now runs smoothly again.

---

> Commented on: 2021-04-20 by @harveyr

I just had this problem for the first time. Host is macOS, client is Ubuntu. The problem seemed to coincide with a large-ish Google Meet on the client machine, which I assume caused a lot of network activity. More troubling than the mouse choppiness was the keyboard activity. It acted like keys were sticking on the client machine. So I'd try to type "Hello" and end up with "Hhhhhhhhhhheloooooooooo."

The problem persisted after the meeting ended. I tried restarting both client and server to no avail. I used the Reload button and hadn't tried manually restarting the processes yet. But just as I arrived here to write this, the problem went away. 🤷 

Edit: This might be my issue https://github.com/debauchee/barrier/pull/679

---

> Commented on: 2021-04-23 by @noveltomatillo8

I'm having this same issue for the first time on my client. Host and client are both W10, WiFi host, ethernet client. The (wireless) mouse cursor lags and skips around for 1-5 seconds every 5-10 seconds no matter what speed I move the cursor but keyboard is totally fine. Switching mice didn't make a difference. Did a clean reinstall of Barrier on both machines to no avail. I only have one band on my router, so changing that isn't an option.

Worked perfectly for a month or so, but now randomly seeing this issue with no changes to either machine that I can think of. Any idea if this is just an underlying bug, or did any fix make a lasting difference?

---

> Commented on: 2021-04-23 by @bbook1997

Somewhat off topic, but I switched over to "Mouse Without Boarders" from Microsoft.   And so far, I've had very little problem with it.  Only once when I had heavy internet traffic did the mouse start lagging, but a quick disconnect / reconnect fixed it immediately.  Also, I had a lot of problems with Barrier connecting my two computers together.   Sometimes having to fiddle with things for several minutes.   I've had zero issues with MWB connecting the two PC's together.   Also, it's a 2-way connection, so I can use either mouse or keyboard.   All in all, much better experience on Windows 10

---

> Commented on: 2021-05-04 by @intositeme

> Somewhat off topic, but I switched over to "Mouse Without Boarders" from Microsoft. And so far, I've had very little problem with it. Only once when I had heavy internet traffic did the mouse start lagging, but a quick disconnect / reconnect fixed it immediately. Also, I had a lot of problems with Barrier connecting my two computers together. Sometimes having to fiddle with things for several minutes. I've had zero issues with MWB connecting the two PC's together. Also, it's a 2-way connection, so I can use either mouse or keyboard. All in all, much better experience on Windows 10

Thanks for the share! Was connecting 2 Windows 10 systems, one on LAN and another on Wifi, and had the jumpy/laggy mouse on the client system. Tried "Mouse Without Boarders" and it didn't have the same kind of lag.

---

> Commented on: 2021-05-04 by @jwallet

I'm still following this thread, but I found out it was better for me to be connected by LAN instead of wirelessly (WAN). I'm sure I will always have issues on wifi because my laptop only connects to the 2.4gHz network, and I also tried to have everything on the same desk (pc (wired), laptop (wifi), router) and I still got issues.

2.4gHz frequency band network can have their beacon signal interfered by many stuff in your house that use this same frequency, like appliances, like microwave or the router next door or even your cellphone,  unlike 5gHz that have less interference. However 5gHz are easily stop by walls, so they reach less far than 2.4gHz band network. 

It might not be what cause your problem, though.
But, if someone wants to try out the app using 5.gHz band network, that would be nice.

---

> Commented on: 2021-05-05 by @clankill3r

I don't think people realize how fast computers are. You don't need 5gHz in order to send the mouse position 60 times a second.

---

> Commented on: 2021-05-31 by @jwallet

No but, like I said, you need a steady connection, I did not talk about the speed, but interference. I'm talking about shitty home router or having too many stuff interfering on your network, that may lower the quality of your wireless signal randomly. I want this software to be accurate at all time, not dropping for a second every 15 minutes or so.

---

> Commented on: 2021-06-10 by @idenc

Even on a 5gHz network, it stutters for me. Only solution was to use LAN

---

> Commented on: 2021-06-21 by @aboode95

To make barrier work over wifi, disable wifi power management:

Run:

    sudo nano /etc/NetworkManager/conf.d/default-wifi-powersave-on.conf

This conf file contains: 

    [connection]
    wifi.powersave = 3

change the value to '2'. And reboot.

These instruction are from [here](https://unix.stackexchange.com/questions/269661/how-to-turn-off-wireless-power-management-permanently).

Big thanks to @noisyshape for mentioning it is a wifi power saving feature issue!

(I am using a laptop with ralink 3090 card. Pretty troublesome card I must say)

---

> Commented on: 2021-08-19 by @sgon00

I am running barrier in three devices: Linux (Debian 10), Windows 10, MacOS (Big Sur).
Linux as the server.
Windows 10 and MacOS as the clients.
I don't have any problems btw Linux and Windows 10.
The lagging issue only happens in MacOS Client. 
To me, the problem is only in MacOS.

---

> Commented on: 2021-08-27 by @intfrr

For me it was solved selecting: "Use relative mouse moves"
<img width="799" alt="Screen Shot 2021-08-27 at 0 25 46_2" src="https://user-images.githubusercontent.com/548549/131082107-452119c6-b1d2-437c-acee-1ef1255cc075.png">

---

> Commented on: 2021-08-27 by @kvnxiao

If you don't need AirDrop, disable the `awdl0` network interface which is for Apple Wireless Direct Link:
```
sudo ifconfig awdl0 down
```

This will help alleviate the choppiness in the mouse movement _a lot_

---

> Commented on: 2021-09-01 by @nocode99

I'm using the server on Big Sur and Ubuntu 20.04 as the client (I also reversed the set up as well) and none of the recommendations helped. I unfortunately cannot test using wired connections but found disabling `Wi-Fi Networking` in Location services to get rid of the issue.  There's a little bit of jitter, but I don't expect the client to be 100% smooth.

Go to *System Preferences -> Security & Privacy -> Location Services* Unlock on the bottom left corner and then select `Details` in System Services. Uncheck `Wi-Fi Networking`

![disable](https://i.imgur.com/9tnFqWL.png)

---

> Commented on: 2021-09-08 by @xQsme

Hi, I believe I'm suffering from this same issue.

I'm running the server on Linux and accessing on a MacOS client.
I've tried every suggestion except for plugging in an ethernet cable (don't have a dongle).

In my case, this issue was very severe, my laptop would freeze constantly, about 2s every 10s, during zoom calls.
After testing my internet stability (https://test.vsee.com/network/index.html), I bought a pretty good Wi-Fi powerline, and my connection became very stable, with low ping, high bandwidth, and no packet loss.

Currently, when listening to music, watching videos, or during zoom calls, I might occasionally get a short freeze (less than 1 sec), where my audio glitches completely.

If i click on the Wi-Fi indicator on MacOS's menu bar, the mouse movement loses it's smoothness for a couple seconds, I've also re-ran the network stability tests while doing this action and I end up with 8-40% packet loss when repeatedly doing this action, though if I disconnect Barrier and use my laptop's touchpad, the connection is flawless.

I'd honestly appreciate it a lot if this issue was resolved, and I'm available to contribute with anything I'm able to.
Thanks.

---

> Commented on: 2021-10-01 by @webdog

I'm experiencing this issue on a hard-wired setup. Each machine is on the same switch so latency is sub 1ms. No error messages reported in the log (logs turned up to highest debug level), but the mouse appears to freeze/jump every 100-500px without fail. Keyboard also suffers from latency. Issues survive a restart, and I've also swapped client/server with no resolution. On each host machine, the keyboard and mouse work fine.

MacOS 11.6 on both machines.

---

> Commented on: 2021-10-17 by @itsCaveMan

I solved my lag by changing client sync to 1000ms

`Configure server --> Advanced server settings --> Check clients every [1000]ms`

![Screenshot 2021-10-17 at 16 43 08](https://user-images.githubusercontent.com/51651537/137632470-0596bdf8-731a-4f6c-9aca-03a397f3b260.png)

I have no lag on my 5g wifi AP, and the odd second here and there on my none-5g wifi AP

Host: MacOS Catalina
Client: Windows 10

**Comment by nengoxx** [comment]
Hi, just to add to this issue, I have a Windows 11 as server, and Linux Mint as client. 
I had a Bluetooth keyboard connected to my Linux laptop previous to the installation of input-leap and had the same issue when I installed it. 

I tried several things, but the problem only disappeared when I disconnected the Bluetooth keyboard from the Linux machine. Apparently it had nothing to do with the network in my case.
